### PR TITLE
test: Code coverage: Transaction outbox service - PR-1

### PR DESCRIPTION
**Pull request by Dimple16**
click up - https://app.clickup.com/t/85zrt3n2j

Screenshot 2023-03-27 at 11 40 39 AM

### CODE REVIEW

File: src/app/core/services/transactions-outbox.service.spec.ts [hunk 1]
```diff
@@ -1,16 +1,244 @@
...
+  it('incrementSingleCaptureCount(): should increment single capture count', () => {
+    transactionsOutboxService.incrementSingleCaptureCount();
+    expect(transactionsOutboxService.singleCaptureCount).toEqual(singleCaptureCountInSession + 1);
+  });
+
+  it('saveQueue(): should save queue', () => {
+    transactionsOutboxService.queue = ['txdzGV1TZEg3'];
+    storageService.set.and.returnValue(Promise.resolve(null));
+    transactionsOutboxService.saveQueue();
+    expect(storageService.set).toHaveBeenCalledWith('outbox', transactionsOutboxService.queue);
```
Comment (rvab) [inline src/app/core/services/transactions-outbox.service.spec.ts:95]: `toHaveBeenCalledOnceWith`

File: src/app/core/services/transactions-outbox.service.spec.ts [hunk 2]
```diff
@@ -1,16 +1,244 @@
...
+  });
+
+  it('saveDataExtractionQueue(): should save data extraction queue', () => {
+    transactionsOutboxService.dataExtractionQueue = ['txdzGV1TZEg3'];
+    storageService.set.and.returnValue(Promise.resolve(null));
+    transactionsOutboxService.saveDataExtractionQueue();
+    expect(storageService.set).toHaveBeenCalledWith(
```
Comment (rvab) [inline src/app/core/services/transactions-outbox.service.spec.ts:102]: same as above

**Review by rvab — APPROVED** [review]
minor comments

File: src/app/core/services/transactions-outbox.service.spec.ts [hunk 3]
```diff
@@ -1,16 +1,244 @@
...
+      dataUrls: fileObjectData1,
+    };
+    transactionsOutboxService.dataExtractionQueue = [entry];
+    spyOn(transactionsOutboxService, 'saveDataExtractionQueue').and.returnValue(Promise.resolve());
+    await transactionsOutboxService.removeDataExtractionEntry(txnData2, fileObjectData1);
+    expect(transactionsOutboxService.saveDataExtractionQueue).toHaveBeenCalledTimes(1);
+    expect(transactionsOutboxService.dataExtractionQueue.length).toEqual(0);
+  });
+
+  it('addDataExtractionEntry():', async () => {
```
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/core/services/transactions-outbox.service.spec.ts]: removed

File: src/app/core/services/transactions-outbox.service.spec.ts (hunk 2 above)
Comment (Dimple16) [reply src/app/core/services/transactions-outbox.service.spec.ts]: fixed both

**Pull request merged**